t/mojo/request_cgi.t

```diff
@@ -267,7 +267,7 @@ $req->parse({
   QUERY_STRING      => '',
   REMOTE_PORT       => '58232',
   HTTP_USER_AGENT   => 'Mojolicious (Perl)',
-  SERVER_PORT       => '13028',
+  SERVER_PORT       => '42',
```

Comment:
Because by changing the value it functions as a test.

* SERVER_PORT should not just work, but it should hold preference over anything else.
* For backwards compat with prior Mojo versions and for servers that don't support that standardized SERVER_PORT, if SERVER_PORT is not present the parsing of HTTP_HOST should occur.

Reply:
Is this still a CGI environment Apache 2.2.14 would generate on its own?

Reply:
@kraih i find it hard to believe that apache would not generate a SERVER_PORT, as it is the standard, but i haven't tested it. If the dumps were taken from that server to generate the tests -- then yes, as it was in the test already.

Reply:
That's a blocker for this PR then.

Reply:
@kraih I'm confused, why would Apache following the standard and us following the standard be a blocker for this PR? Could you explain?

Reply:
You change one of our previous Apache cases to test something new, without ensuring that the previous case still works. That's not acceptable.

Reply:
@kraih What do you mean when you say "without ensuring that the previous case still works"? I didn't modify any of the functionality in the previous test, I just expanded on it to accommodate new functionality. If you run this test against the code without the patch, it'll fail. Because prior to the patch `SERVER_PORT`, if different from `HTTP_HOST` didn't do anything.

Previously you had multiple tests that were testing `HTTP_HOST`

* I changed one of them to have a discrepancy between `HTTP_HOST` and `SERVER_PORT`, and made sure that `SERVER_PORT` (the standard option) took precedence.
* If both exist and they have the same value it's moot. This is already tested here (https://github.com/mojolicious/mojo/blob/7cf399f4bf146a5cc174bc01b92a3cd2ba8d3d65/t/mojo/request_cgi.t#L428)
* If only `HTTP_HOST` exists the patch does nothing new. This is tested here. https://github.com/mojolicious/mojo/blob/7cf399f4bf146a5cc174bc01b92a3cd2ba8d3d65/t/mojo/request_cgi.t#L110

I can certainly copy and paste the entire test if that's the ask, but I'm not sure what we're trying to accomplish.